shanoir-ng-datasets/src/main/java/org/shanoir/ng/datasetfile/DatasetFileApiController.java

```diff
@@ -0,0 +1,109 @@
+package org.shanoir.ng.datasetfile;
+
+import java.io.File;
+import java.time.LocalDateTime;
+import javax.validation.Valid;
+
+import org.apache.commons.io.FileUtils;
+import org.shanoir.ng.datasetfile.service.DatasetFileApi;
+import org.shanoir.ng.datasetfile.service.DatasetFileService;
+import org.shanoir.ng.shared.exception.ErrorModel;
+import org.shanoir.ng.shared.exception.RestServiceException;
+import org.shanoir.ng.shared.service.DicomServiceApi;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.multipart.MultipartFile;
+
+import io.swagger.annotations.ApiParam;
+
+@Controller
+public class DatasetFileApiController implements DatasetFileApi {
+
+	@Autowired
+	DatasetFileService datasetFileService;
+
+	@Autowired
+	@Qualifier("stowrs")
+	DicomServiceApi stowRsService;
+
+	@Autowired
+	@Qualifier("cstore")
+	DicomServiceApi cStoreService;
+
+	@Value("${dcm4chee-arc.protocol}")
+	private String dcm4cheeProtocol;
+
+	@Value("${dcm4chee-arc.host}")
+	private String dcm4cheeHost;
+
+	@Value("${dcm4chee-arc.port.web}")
+	private String dcm4cheePortWeb;
+
+	@Value("${dcm4chee-arc.dicom.web}")
+	private boolean dicomWeb;
+
+	@Override
+	public 	ResponseEntity<DatasetFile> saveNewDatasetFile(
+			@ApiParam(value = "datasetfile to create", required = true) @RequestBody DatasetFile datasetFile,
+			BindingResult result) throws RestServiceException  {
+		try {
+			// Save file
+			if (datasetFile.isPacs()) {
+				String oldPath = datasetFile.getPath();
+				String newPath = oldPath.replaceAll("http(.*)wado\\?", dcm4cheeProtocol + dcm4cheeHost + ":" + dcm4cheePortWeb + "/wado?");
```

Comment:
Hi Jean-Come, this translation, that can vary a lot between systems, should be configurable outside please.
Something like a wadoTranslateRule per import, depending where the data are coming from, at least put it into
the application.yml to have it outside the code?

Reply:
I'll do this totally differently I think, this is not quite logic to use a regex here as we have all the informations in application.yml already.